packages/components/BuyerDelivery.tsx

```diff
@@ -1,5 +1,329 @@
-import { Box } from '@chakra-ui/react';
+import {
+  Box,
+  Button,
+  Collapse,
+  FormControl,
+  FormControlProps,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  FormLabelProps,
+  Input,
+  Radio,
+  RadioGroup,
+  Select,
+  Stack,
+  Text,
+  TextProps,
+} from '@chakra-ui/react';
+import { FormikErrors, FormikValues, useFormik } from 'formik';
+
+import UFS from '@packages/config/ufs';
+import { DeliveryType } from '@packages/enums/DeliveryType';
+
+import InputMask, { cepMask } from './InputMask';
+
+const titleStyle: TextProps = {
+  marginTop: '1em',
+  fontSize: '25px',
+  fontWeight: 'bold',
+  color: 'black',
+};
+
+const fieldsetStyle: FormControlProps = {
+  border: '1px solid #ccc',
+  padding: '1em',
+  borderRadius: '0.3em',
+};
+
+const fieldsetLabelStyle = {
+  fontSize: '18px',
+  fontWeight: 'bold',
+  color: 'gray.600',
+};
+
+const radioLabelStyle: TextProps = {
+  fontSize: '14px',
+  fontWeight: 'bold',
+  color: 'gray.600',
+};
+
+const labelStyle: FormLabelProps = {
+  fontSize: '14px',
+  color: 'gray.600',
+};
+
+const inputStyle = {
+  borderColor: 'gray.400',
+  borderRadius: '4px',
+  bgColor: 'gray.300',
+};
+
+function validateFieldFilled(
+  value: string,
+  fieldName: string,
+  size: number,
+): string | undefined {
+  if (!value) {
+    return `${fieldName} é requerido(a).`;
+  }
+
+  if (value.length < size) {
+    return `${fieldName} deve conter no mínimo ${size} caracteres.`;
+  }
+
+  return undefined;
+}
+
+interface FormValues extends FormikValues {
+  deliveryType: DeliveryType;
+  address: string;
+  number: string;
+  complement: string;
+  district: string;
+  cep: string;
+  city: string;
+  uf: string;
+}
 
 export default function BuyerDelivery() {
-  return <Box>Forma de Entrega</Box>;
+  const formik = useFormik<FormValues>({
+    initialValues: {
+      deliveryType: DeliveryType.MAIL,
+      address: '',
+      number: '',
+      complement: '',
+      district: '',
+      cep: '',
+      city: '',
+      uf: '',
+    },
+    validate: (values: FormValues) => {
+      const errors: FormikErrors<FormValues> = {};
+      const fields = [
+        { key: 'address', fieldName: 'Logradouro', size: 10 },
+        { key: 'district', fieldName: 'Bairro', size: 5 },
+        { key: 'cep', fieldName: 'CEP', size: 9 },
+        { key: 'city', fieldName: 'Cidade', size: 10 },
+        { key: 'uf', fieldName: 'Estado', size: 2 },
+      ];
+
+      if (values.deliveryType === DeliveryType.MAIL) {
+        fields.forEach(({ key, fieldName, size }) => {
+          errors[key] = validateFieldFilled(values[key], fieldName, size);
+
+          if (errors[key] === undefined) {
+            delete errors[key];
```

Comment:
essa lógica de deletar se for indefinido ficou um pouquinho estranha 😅 . O que acha de fazer ao contrário (escrever se for uma string)?

Reply:
Acho que dá certo também. 

O negócio é que o formik parace que não faz o submit se houver alguma chave no objeto error. Mesmo que esteja com valor de indefinido.